app/src/main/java/org/novinger/jason/demoreader/ListActivity.java

```diff
@@ -1,47 +1,70 @@
+import android.util.Log;
 import android.view.Menu;
 import android.view.MenuItem;
 import android.view.View;
+import android.view.ViewPropertyAnimator;
 import android.widget.AdapterView;
 import android.widget.ArrayAdapter;
 import android.widget.ListView;
 
+import com.google.gson.Gson;
+
+import org.novinger.jason.demoreader.datamodels.Article;
+
+import java.io.IOException;
 import java.util.ArrayList;
-import java.util.Collections;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
 
 public class ListActivity extends AppCompatActivity {
 
+    private final static int qty = 50;
+    private final static String feedUrl = "http://www.dailydot.com/api/v1/content/article/";
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_list);
 
+        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
+        StrictMode.setThreadPolicy(policy);
+        Article[] articles;
+        try {
+            articles = getFeedItems(getFeedUrl());
+        } catch (IOException ex) {
+            articles = new Article[0];
+            Log.d("Exception", ex.toString());
+        }
+
         final ListView listview = (ListView) findViewById(R.id.listview);
-        String[] values = new String[] {
-                "Jason", "Cindy", "Hannah", "Leah", "Kirsten", "Matt", "Elle", "Trevor", "Alex",
-                "Eli", "Danika", "Lori", "Wyatt", "Chad"
-        };
-        final ArrayList<String> list = new ArrayList<String>();
-        Collections.addAll(list, values);
+        final ArrayList<String> list = new ArrayList<>();
+        for (Article article : articles) {
+            list.add(article.toString());
+        }
 
         final ArrayAdapter adapter = new ArrayAdapter(this, android.R.layout.simple_list_item_1, list);
         listview.setAdapter(adapter);
         listview.setOnItemClickListener(new AdapterView.OnItemClickListener() {
             @Override
             public void onItemClick(AdapterView<?> parent, final View view, int position, long id) {
                 final String item = (String) parent.getItemAtPosition(position);
-                view.animate().setDuration(500).alpha(0)
-                    .withEndAction(new Runnable() {
-                        @Override
-                        public void run() {
-                            list.remove(item);
-                            adapter.notifyDataSetChanged();
-                            view.setAlpha(1);
-                        }
-                    });
+                final ViewPropertyAnimator viewPropertyAnimator = view.animate().setDuration(100).alpha(0);
+                viewPropertyAnimator.withEndAction(new Runnable() {
+                    @Override
+                    public void run() {
+                        list.remove(item);
+                        list.trimToSize();
```

Comment:
yeah that's exactly when i use System.gc - lets say I'm doing some crazy bitmap manipulations, but i think I'm managing memory correctly - say we're on a modern tablet with a large heap - we decode the bitmap (memory), scale it (memory, memory, memory), then we made a BitmapRegionDecode (because we only want to show a part of it) (MEMORY!!!), then we return the small piece we want to show (memory), _then_ we exit the stack (end of function, as we `return` the small bitmap) - we look memory monitor it probably shows 118 being used out of the 120MB we have available (because of all that bitmap allocation), but since that memory is available for GC (since we left that method's stack), if we now run a System.gc (to emulate what would happen if the system needed more memory than was available at the moment), we'd probably see that monitor dip back down to single or double digits...  yay!

that all said, even if we do get to reclaim it, there's still heap limits for devices: http://stackoverflow.com/a/9940415/429430, so while we may think "oh this is temporary (on the stack), no problem", if we actually run out of memory before we're done, we'll crash - even if we `catch` the OOME, we'll still fail to create that small bitmap we wanted...  memory management is a bitch :/